app/src/main/java/com/locked/shingranicommunity/event/EventListViewModel.kt

```diff
@@ -162,7 +162,9 @@ class EventListViewModel @Inject constructor(
         }
 
         fun openMap() {
-            navigation.navigateToMap(eventItem.address!!)
+            eventItem.address?.let {
+                navigation.navigateToMap(it)
+            }
```

Comment:
revert this back to `navigation.navigateToMap(eventItem.address!!)`
instead put your conditional logic inside `init {}` to show/not-show the map button like this:
```kotlin
  init {
            ...
            eventItem.address?.let {
                data.showMap.value = true
            }
  }

    data class ItemData(
        val name: MutableLiveData<String> = MutableLiveData(""),
        val desc: MutableLiveData<String> = MutableLiveData(""),
        val type: MutableLiveData<String> = MutableLiveData("Event"),
        val date: MutableLiveData<String> = MutableLiveData(""),
        val time: MutableLiveData<String> = MutableLiveData(""),
        val attendees: MutableLiveData<Int> = MutableLiveData(0),
        val showAccept: MutableLiveData<Boolean> = MutableLiveData(true),
        val showReject: MutableLiveData<Boolean> = MutableLiveData(true),
        val showDelete: MutableLiveData<Boolean> = MutableLiveData(true),
        val showShare: MutableLiveData<Boolean> = MutableLiveData(true),
        val showMap: MutableLiveData<Boolean> = MutableLiveData(false),
        val showDeleteConfirmation: MutableLiveData<Boolean> = MutableLiveData(false)
    )

```

Notice in the `ItemData` class, the default value for `showMap` to be `false`.